### ISSUE TITLE
New Input Processors

### ISSUE THREAD

**Issue by Shute052** [Leafwing-Studios/leafwing-input-manager#491]
Add the input processors, similar to those in [Unity](https://docs.unity3d.com/Packages/com.unity.inputsystem@1.0/manual/Processors.html), and replace the existing ones

## Processor Types

- Clamp
- Filter: for all values, only negative values, only positive values, or no passing values
- Sensitivity
- Inversion
- Dead Zone: circle, square, rounded square

**Comment by alice-i-cecile** [comment]
Oh interesting: that's a sensible framing. We should try to split this out into distinct work from #490 though: overly large refactor steps tend to be much harder to work with.

**Comment by Shute052** [comment]
yeah, I've been trying to split them out. #490 has made some progress, but I haven't pushed the changes yet.

**Comment by Shute052** [comment]
Updated the description

**Comment by Shute052** [comment]
Updated the description by combining Filters into Clamps and removing user-defined processors reliant on Box<dyn Fn>, as they are not serializable (use of `serde_traitobjects` and `serde_closures` requires Rust nightly)
